### PR TITLE
Updated 'AppleProvider.cs' for supporting other TokenTypes than just AccessToken

### PR DESCRIPTION
**Pull request by themine702**
A small change to allow the use of other tokens than AccessToken with the AppleProvider

### CODE REVIEW

File: src/Auth/Providers/AppleProvider.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ public AppleProvider()
             this.AddScopes(DefaultEmailScope);
         }
 
-        public static AuthCredential GetCredential(string accessToken) => GetCredential(FirebaseProviderType.Apple, accessToken, OAuthCredentialTokenType.AccessToken);
+        public static AuthCredential GetCredential(string Token, OAuthCredentialTokenType TokenType) => GetCredential(FirebaseProviderType.Apple, Token, TokenType);
```
Comment (StephenHodgson) [inline src/Auth/Providers/AppleProvider.cs:12]: improper PascalCasing on parameter names
```suggestion
        public static AuthCredential GetCredential(string token, OAuthCredentialTokenType tokenType) => GetCredential(FirebaseProviderType.Apple, token, tokenType);
```

Comment (StephenHodgson) [reply src/Auth/Providers/AppleProvider.cs:12]: Could we also set the default `OAuthCredentialTokenType` to `AcessToken`? That way it's not a breaking change.

Comment (StephenHodgson) [inline src/Auth/Providers/AppleProvider.cs:12]:
```suggestion
        public static AuthCredential GetCredential(string token, OAuthCredentialTokenType tokenType = OAuthCredentialTokenType.AccessToken) => GetCredential(FirebaseProviderType.Apple, token, tokenType);
```

**Comment by bezysoftware** [comment]
Is this something that's unique for Apple provider? Should we not allow the same thing for all the other providers? If so, can we leverage this new c# feature https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/static-virtual-interface-members ?

**Pull request closed without merging**